### PR TITLE
Initial support for HistGradientBoostingClassifier / HistGradientBoostingRegressor

### LINKED ISSUE #386: Support sklearn HistGradientBoosting models
[HistGradientBoostingRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html#sklearn.ensemble.HistGradientBoostingRegressor)

[HistGradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html#sklearn.ensemble.HistGradientBoostingClassifier)

---

I'm very keen to tackle this issue myself. I hope that's ok with you @hcho3

### LINKED ISSUE #434: Support for HistGradientBoostingClassifier
Hi @hcho3,

As the title suggests, is support for Scikit-learn's [hist gradient boosting classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html) on the roadmap as those already implemented [here](https://treelite.readthedocs.io/en/latest/tutorials/import.html?highlight=Scikit#importing-scikit-learn-models)?

If not, is there a reason why? Or the wrapper to the "standard" gradient boosting classifier is enough to wrap its hist variant as well?

### PR DESCRIPTION
**Pull request by hcho3**
Closes #434 
Closes #386

Support `HistGradientBoostingClassifier` and `HistGradientBoostingRegressor` from scikit-learn.

Limitation: `HistGradientBoostingClassifier` with `n_classes > 2` (i.e. multi-class classification) is not yet supported. This is because Treelite models can hold only scalars in the `global_bias` field (also see the Review comment below):
https://github.com/dmlc/treelite/blob/c59fa9b2fe746d0c0324472e5d1cc1258d1ede56/include/treelite/tree.h#L702-L708

I'd like to revise the Treelite model object so that it can hold a vector `global_bias`. #445

### CODE REVIEW

File: include/treelite/frontend.h [hunk 1]
```diff
@@ -220,11 +220,82 @@ std::unique_ptr<treelite::Model> LoadSKLearnGradientBoostingRegressor(
...
+ * \param value value[i][k] stores the leaf output of node k of the i-th tree. This is only defined
+ *              if node k is a leaf node.
+ * \param n_node_samples n_node_samples[i][k] stores the number of data samples associated with
+ *                       node k of the i-th tree.
+ * \param gain gain[i][k] stores the gain (reduction of the loss function) associate with node k of
+ *             the i-th tree. This is only defined if node k is an internal (non-leaf) node.
+ * \param baseline_prediction Baseline predictions for outputs. At prediction, margin scores will be
+ *                            adjusted by this amount before applying the post-processing (link)
+ *                            function. Required shape: (1,) for binary classification;
+ *                            (n_classes,) for multi-class classification
```
Comment (hcho3) [inline include/treelite/frontend.h:299]: Note that `baseline_prediction` will be of shape `(n_classes,)` when `n_classes > 2`. This is currently not supported, since `global_bias` must be a scalar.
